format/spec.md

```diff
@@ -584,8 +589,8 @@ The schema of a manifest file is a struct called `manifest_entry` with the follo
 | _optional_ | _optional_ | _optional_ | **`110  null_value_counts`**      | `map<121: int, 122: long>`                                                  | Map from column id to number of null values in the column                                                                                                                                                          |
 | _optional_ | _optional_ | _optional_ | **`137  nan_value_counts`**       | `map<138: int, 139: long>`                                                  | Map from column id to number of NaN values in the column                                                                                                                                                           |
 | _optional_ | _optional_ | _optional_ | **`111  distinct_counts`**        | `map<123: int, 124: long>`                                                  | Map from column id to number of distinct values in the column; distinct counts must be derived using values in the file by counting or using sketches, but not using methods like merging existing distinct counts |
-| _optional_ | _optional_ | _optional_ | **`125  lower_bounds`**           | `map<126: int, 127: binary>`                                                | Map from column id to lower bound in the column serialized as binary [1]. Each value must be less than or equal to all non-null, non-NaN values in the column for the file [2]                                     |
-| _optional_ | _optional_ | _optional_ | **`128  upper_bounds`**           | `map<129: int, 130: binary>`                                                | Map from column id to upper bound in the column serialized as binary [1]. Each value must be greater than or equal to all non-null, non-Nan values in the column for the file [2]                                  |
+| _optional_ | _optional_ | _optional_ | **`125  lower_bounds`**           | `map<126: int, 127: binary>`                                                | Map from column id to lower bound in the column serialized as binary [1]. Each value must be less than or equal to all non-null, non-NaN values in the column for the file [2]. See [7] for`geometry` and [8] for `geography`.  |
+| _optional_ | _optional_ | _optional_ | **`128  upper_bounds`**           | `map<129: int, 130: binary>`                                                | Map from column id to upper bound in the column serialized as binary [1]. Each value must be greater than or equal to all non-null, non-Nan values in the column for the file [2]. See [9] for `geometry` and [10] for `geography`. |
```

Comment:
Yes. This accommodates this (westmost could be > eastmost). There was a sentence explicitly highlights it but I think @szehon-ho removed it.

Reply:
@jiayuasu trying to look back on the commit history and cant find it.  I think it was always this one:  this is a point: X = westernmost bound of all geometries in file, Y = northernmost bound of all geometries in file, Z is min value for all component points of all geometries in the file, M is min value of all component points of all geometries in the file. See Appendix D for encoding.   (covered in the note [9]).  Let me know if something else need to be added though.

Reply:
@szehon-ho 

I found it from my PR: https://github.com/szehon-ho/iceberg/pull/2/files

Can you figure out a way to add it back? 


> The concepts of westernmost and easternmost values are explicitly introduced to address cases involving antimeridian crossing, where the `lower_bound` may be greater than `upper_bound`.

Reply:
whoops, must have accidentally removed as there are too many versions.  Put it back, and moved the other common note about 180/90 bound to this point as well.  This is for geography only right?  (the new point is for geography type)

Reply:
@szehon-ho I suggest we put it back for both `Geography` and `Geometry`. As we agreed, `Geometry` allows all types of CRS: `GeographicCRS`, `ProjectedCRS` and `LocalCRS`, while we treat all edges as linear. This introduces `antimeridian crossing` to `Geometry` as well so it would make sense to allow `lower_bound ` greater than `upper_bound`

Reply:
@jiayuasu beginner questions to learn:

What is the difference between a Geography type with planar edge algorithm and Geometry in this case ?  (We dont have planar edge algorithm type, but just assuming we model that way). I thought initially the difference is Geometry doesnt have the wraparound and Geography does.

And also in this case, should I just collapse 7+8 and 9 + 10 then?  I guess the westernmost/ easternmost concept we put in 8/10 apply to Geometry as well then? (I assume thats the same as anti-meridian crossing).

Reply:
@szehon-ho We do not allow planar edge algorithm in Geography type.

Geography type only allows GeographicCRS and one of the given non-planar edge algorithms.
Geometry type allows all CRSs but only allows the planar edge algorithm.

The anti-meridian crossing issue can happen to both types so I think it makes sense to allow them in the bbox of both types.

And, yes, I think you should collapse 7+8 and 9+10.

Reply:
OK got it , so the difference of Geometry and Geography (other than planar edge algorithm), is the CRS are also different, so it makes more sense to have two different types.

I collapsed 7+8 and 9+10.  I also extracted out the point about -180/180 (https://github.com/apache/iceberg/pull/10981#discussion_r1881245122) into the common point, but note its only refer to geography.  Please double check again, thanks again.

Reply:
@paleolimbot i apologize, did you mean add the following sentence?  

```
Whereas the case where xmin <= xmax requires that x >= xmin AND x <= xmax to imply potential intersection, the case where xmin > xmax implies a potential intersection if x >= xmin OR x <= xmax.
```

Do you have any suggestion how to phrase the lower_bound and upper_bound (points 7, 8) overall?  Im struggling a little bit to think how to phrase it holistically

Reply:
@paleolimbot I guess 9 has a typo, it should be `easternmost`

```
9. For the X value only, this is permitted to be the easternmost longitude if the CRS is a geographic CRS. See Appendix D for encoding.
```

I think we should still have it otherwise filter pushdown on antimeridian crossing objects in `Geometry` type is impossible. Unless we chop the original geometry to parts, which is undesirable in a database system.

Reply:
Yea to @paleolimbot 's point, it seems hard for implementation to know whether a custom CRS is geographic or not as of now, so I imagine the Geometry type pushdown will have to deal with antimeridian crossing (xmax < xmin) scenario for all CRS, is that correct then?  

Ie, can we leave out the geographic crs part of points 7, 9?  In that case, the only difference with 8, 10 is that the Geography mention northernmost / southernmost as well.

Reply:
Hi folks, I think mostly makes sense, but I still struggle a little bit, would appreciate some help:

For geometry we say `min value` and `max value` of all geometries in the file, but then say xmin > xmax is possible, which seems contradictory.    Is there a better term then ?  (unfortunately we cant use westernmost / easternmost)

Reply:
OK the description is then the same as field name :) , but its fine I guess

Reply:
I made another attempt, also tried to refactor out common language in the suggestion into common point to reduce from 5 to 3 points.

```
7. `geometry` and `geography`: this is a point: X, Y, Z, and M are the lower / upper bound of all component points of all geometries in file.
8. `geometry`: For the X value only, the lower_bound's X value (xmin) may be greater than the upper_bound's X value (xmax), in which case a geometry in the file may match if it contains an X such that `x >= xmin` OR `x <= xmax`. This definition is agnostic to coordinate system.
9. `geography`: For the X and Y values, the `lower_bound`'s X (westernmost) and Y (northernmost) values may be greater than the `upper_bound`'s X (easternmost) and Y (southernmost) values. The canonical ranges for these points in the coordinate system is [-180 180] for west-east and [-90 90] for the south-north.
```

Let me know how it sounds?

Also, do we need westernmost, northernmost, etc language for geography now that we have a mathematical definition?  I was wondering whether we can even re-use the geometry math definition for x, and extend it to y-axis, so that we have even less text.  One reason is we introduce westernmost, northernmost, but don't really define them.